src/serac/physics/solid_functional.hpp

```diff
@@ -184,6 +279,29 @@ class SolidFunctional : public BasePhysics {
     mesh_.NewNodes(*mesh_nodes, true);
   }
 
+  template < typename T >
+  QuadratureData<T> createQuadratureDataBuffer(T initial_state) {
+    //const mfem::FiniteElement&   el = *(displacement_.space().GetFE(0));
+    //const mfem::IntegrationRule& ir = mfem::IntRules.Get(el.GetGeomType(), el.GetOrder() * 2);
+    //constexpr auto flags = mfem::GeometricFactors::COORDINATES | mfem::GeometricFactors::JACOBIANS;
+    //auto geom = mesh_.GetGeometricFactors(ir, flags);
+
+    constexpr auto Q                                 = order + 1;
+    constexpr auto num_quadrature_points_per_element = (dim == 2) ? Q * Q : Q * Q * Q;
```

Comment:
A note that this is assuming hexes or quads with tensor product integration rules would be nice. I know it is also required by other parts of the code, but it would help readers understand why this is hard coded.